lib/inertia_rails/rspec.rb

```diff
@@ -0,0 +1,117 @@
+require "rspec/core"
+require "rspec/matchers"
+
+module InertiaRails
+  module RSpec
+    class InertiaRenderWrapper
+      attr_reader :view_data, :props, :component
+    
+      def initialize
+        @view_data = nil
+        @props = nil
+        @component = nil
+      end
+    
+      def call(params)
+        set_values(params)
+        @render_method&.call(params)
+      end
+    
+      def wrap_render(render_method)
+        @render_method = render_method
+        self
+      end
+    
+      protected
+    
+      def set_values(params)
+        @view_data = params[:locals].except(:page)
+        @props = params[:locals][:page][:props]
+        @component = params[:locals][:page][:component]
+      end
+    end
+
+    module Helpers
+      def inertia
+        raise 'Inertia test helpers aren\'t set up! Make sure you add inertia: true to describe blocks using inertia tests.' unless inertia_tests_setup?
+
+        @_inertia_render_wrapper
+      end
+
+      def expect_inertia
+        expect(inertia)
+      end
+
+      def inertia_wrap_render(render)
+        @_inertia_render_wrapper = InertiaRenderWrapper.new.wrap_render(render)
+      end
+
+      protected 
+      
+      def inertia_tests_setup?
+        @_inertia_render_wrapper.present?
+      end
+    end
+  end
+end
+
+RSpec.configure do |config|
+  config.include ::InertiaRails::RSpec::Helpers
+
+  config.before(:each, inertia: true) do
+    new_renderer = InertiaRails::Renderer.method(:new)
+    allow(InertiaRails::Renderer).to receive(:new) do |component, controller, request, response, render, named_args|
+      new_renderer.call(component, controller, request, response, inertia_wrap_render(render), named_args)
+    end
+  end
+end
+
+RSpec::Matchers.define :have_props do |expected_props|
+  match do |inertia|
+    expect(inertia.props).to eq expected_props
+  end
+
+  failure_message do |inertia|
+    "expected inertia props to receive #{expected_props}, instead received #{inertia.props || 'nothing'}"
+  end
+end
+
+RSpec::Matchers.define :include_props do |expected_props|
+  match do |inertia|
+      expect(inertia.props).to include expected_props
+  end
+
+  failure_message do |inertia|
+    "expected inertia props to include #{expected_props}, instead received #{inertia.props || 'nothing'}"
+  end
+end
```

Comment:
oooh, now i get the difference between have_props and include_props.... makes sense! i thought they were equivalent at first glance. what do you think about changing it to `have_exact_props`?

Reply:
Yeah I definitely see what you're saying; I'm sure you wouldn't be the only person thinking that way.

Do you think it should be an outright change or an alias?